<a name="readme-top"></a>
<!-- PROJECT SHIELDS -->
<br />
<div align="center">
  <a href="https://github.com/allyvaz/Capstone-project">
    <img src="https://microbenotes.com/wp-content/uploads/2023/05/Alleles.jpg" alt="Logo" width="80" height="80">
  </a>

<h3 align="center">Variants of <i>TP53</i> causing Breast Cancer</h3>

  <p align="center">
    Project Overview
  </p>
  <p align ="justify">
    <b>Areas of Interest:</b> 
    <br/>1. Variants of <i>TP53</i> causing Breast Cancer.
    <br/>2. Variant allele frequency of <i>TP53</i>.
    <br/>3. Longitudinal study of <i>TP53</i> variants.
    <br/>4. Exploration of <i>TP53</i> variant interactions with other genes or pathways implicated in Breast Cancer development.
    <br/><b>Problem Statement: </b>
    <br/>In the realm of cancer research, accurately identifying mutations (also known as variants) from next-generation sequencing of tumor samples poses a significant challenge. Variants exhibit diverse characteristics within the dataset, and both biological and technical variability complicate the process for variant callers. Various metrics, such as sequencing depth around the variant, the count of reads supporting the reference or mutated allele, the quality associated with these reads, and the confidence level of the tools in making specific calls, are crucial for assessing variants. This project will specifically explore the utility of one such metric: Variant Allele Frequency (VAF), in addressing these challenges.
    <br/><b>Proposed Data Science strategy:</b>
    <br/> 1. Prioritize <i>TP53</i> variants based on their predicted functional impact, frequency in the dataset, and relevance to breast cancer pathogenesis.
    <br/>2. Utilize machine learning algorithms, such as random forests or gradient boosting, to rank  <i>TP53</i> variants according to their likelihood of driving breast cancer progression or treatment response. 
    <br/><b>Impact of this solution:</b>
    <br/>1. Improved Prioritization of Variants: Machine learning models can consider multiple features simultaneously, including variant type, allele frequency, and clinical data, to prioritize <i>TP53</i> variants more accurately. This can help researchers focus their efforts on investigating variants most likely to be clinically relevant.
    <br/>2. Identification of Biomarkers: By identifying <i>TP53</i> variants associated with disease progression or treatment response, this approach can lead to the discovery of biomarkers that could be used for patient stratification, prognosis prediction, and treatment selection like Personalized Treatment Strategies.
    <br/><b>Description of the dataset:</b>
    
  </p>
   
